zkvm/src/txlog.rs

```diff
@@ -118,3 +167,137 @@ impl Entry {
         }
     }
 }
+
+impl MerkleTree {
+    /// Constructs a new MerkleTree based on the input list of entries.
+    pub fn new(list: &[Entry]) -> Option<MerkleTree> {
+        if list.len() == 0 {
+            return None;
+        }
+        let t = Transcript::new(b"ZkVM.txid");
+        Some(MerkleTree {
+            size: list.len(),
+            root: MerkleNode::build_tree(t, list),
+        })
+    }
+
+    /// Builds a proof of inclusion for entry at the given index for the Merkle tree.
+    pub fn proof(&self, index: usize) -> Result<Vec<MerkleNeighbor>, VMError> {
+        if index >= self.size {
+            return Err(VMError::InvalidMerkleProof);
+        }
+        let t = Transcript::new(b"ZkVM.txid");
+        let mut result = Vec::new();
+        self.root.subproof(t, index, self.size, &mut result);
+        Ok(result)
+    }
+}
+
+impl MerkleNode {
+    fn build_tree(mut t: Transcript, list: &[Entry]) -> Self {
+        match list.len() {
+            0 => {
+                let mut leaf = [0u8; 32];
+                TxID::empty(t, &mut leaf);
+                return MerkleNode::Leaf(leaf);
+            }
+            1 => {
+                let mut leaf = [0u8; 32];
+                TxID::leaf(t, &list[0], &mut leaf);
```

Comment:
let's move `empty/leaf/node` functions from under TxID to MerkleTree.

Also since `build_tree` is not even taking `&self`, we can move it under MerkleTree too.